src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java

```diff
@@ -72,7 +74,15 @@ private static SymbolLookup makeSystemLookup() {
     }
 
     private static SymbolLookup makeWindowsLookup() {
-        Path system32 = Path.of(System.getenv("SystemRoot"), "System32");
+        @SuppressWarnings("removal")
+        String systemRoot = AccessController.doPrivileged(new PrivilegedAction<String>() {
+            @Override
+            public String run() {
+                return System.getenv("SystemRoot");
+            }
+        });
```

Comment:
We can use `sun.security.action.GetPropertyAction.privilegedGetProperty` here for simplicity. See e.g. https://github.com/openjdk/panama-foreign/blob/foreign-memaccess%2Babi/src/java.base/share/classes/jdk/internal/foreign/CABI.java#L50 where it's used as well.

Reply:
Never mind, that is for System properties. Here we're reading an environment variable.

Reply:
I mean create such a class by copying what the other `sun.security.action.Get*Action` classes do.